src/components/Modal/__tests__/modal-snapshot-tests.jest.js

```diff
@@ -0,0 +1,58 @@
+import React from "react";
+import renderer from "react-test-renderer";
+import Modal from "../Modal";
+import { ModalContent, ModalFooter, ModalHeader } from "components";
+
+const baseProps = {
+  id: "modal-id",
+  show: true,
+  title: "Modal Title",
+  onClose: () => {}
+};
+
+const content = <p>content</p>;
+const footerContent = <p>footer</p>;
+
+describe("Modal renders correctly", () => {
+  it("with base props", () => {
+    const tree = renderer.create(<Modal {...baseProps}>{content}</Modal>).toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+
+  it("with Header", () => {
+    const tree = renderer
+      .create(
+        <Modal {...baseProps}>
+          <ModalHeader title={"ModalHeader Title"} />
+          {content}
+        </Modal>
+      )
+      .toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+
+  it("with Footer", () => {
+    const tree = renderer
+      .create(
+        <Modal {...baseProps}>
+          {content}
+          <ModalFooter>{footerContent}</ModalFooter>
+        </Modal>
+      )
+      .toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+
+  it("with all parts", () => {
```

Comment:
I think we need to check all/most props permutations:
 - classNames
 - title
 - descriptions
 - etc. - all props
 I really recommend to use snapshotDiff for decrease snapshot size and improve readability of snapshots.